src/index.ts

```diff
@@ -29,32 +29,45 @@ const gateURI = 'http://localhost:3000';
  *               to the webapp's backend
  *
  */
+
 // instantation, everything before the return callback runs only once
 export default function gatelog(projectID: string, apiKey: string) {
-    // const newAuth = new AuthVerification(gateURI, projectID, apiKey);
-
-    // const validate = newAuth.endpointValidation(req, res, next);
-    // const verify = newAuth.keyVerification(req, res, next);
-
-    // run the API Key verification process when gateLogger is instantiated
-    try {
-        // await validate();
-        // await verify();
-    } catch (err) {
-        return err;
-    }
-
-    // const postQuery = new PostQuery(gateURI, projectID);
-
-    // every time a request is processed in the user's backend
-    return async (req: Request, res: Response, next: NextFunction) => {
-        // reassign res.end in order to allow logger functionality before
-        // a response is sent back the client
-        // eslint-disable-next-line arrow-body-style
-        res.end = () => {
-            // postQuery.post(req, res, next);
-            return res;
-        };
+    const newAuth = new AuthVerification(gateURI, projectID, apiKey);
+
+    const validate = newAuth.validation;
+    const verify = newAuth.verification;
+
+    // run the API Key verification process when gatelog is instantiated
+    validate().catch((err) => err);
+    verify().catch((err) => err);
+
+    // every time a request is processed in the user's backend,
+    // this express middleware callback will run
+    return (req: Request, res: Response, next: NextFunction) => {
+        // calls initial timestamp of request's beginning,
+        // this will depend on where logger is placed in middleware chain
+        const timestamp: number = new Date().valueOf();
+
+        // runs logger's functionality upon response being sent back to client
+        res.on('finish', async (): Promise<void> => {
+            // calls timestamp of request's end
+            const loggedOn: number = new Date().valueOf();
+
+            // stores time between request's beginning and end
+            const latency: number = loggedOn - timestamp;
+
+            // passes into PostQuery class all query data along with recorded timestamp, loggedOn, and latency
+            const postQuery = new PostQuery(gateURI, projectID, {
```

Comment:
What is the benefit of create a new `PostQuery` object? Are we able to just call a single `post` function as opposed to instantiated a PostQuery class and then calling post()?

Reply:
changed to an exported function instead and tested, everything is working